### PR TITLE
Allow field aliasing with Serde.

### PR DESCRIPTION
**Pull request by calavera**
Hi! :wave: 

I have the misfortune of naming a field in a JSON payload `static`. This library transforms that field name into `field_static`. With this change, my payloads are not deserialized correctly :pensive:

Keeping compatibility between two different formats will always be hard, but the transition can be slightly better for fields that are name like reserved Rust keywords.

This PR introduces a private field `wire_name` in the Field definition to store the original name that the definition's author gave to a name. If that name doesn't match the final field name, we add an alias for serde to consider during deserialization.

I've extracted the code that writes serde field attributes into a separated function so it's decoupled from the rest of the code that generates the field's code.

Let me know if you have any suggestion to make this code better :raised_hands: 

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: protobuf-codegen/src/field/mod.rs [hunk 1]
```diff
@@ -558,6 +558,7 @@ pub(crate) struct FieldGen<'a> {
     pub rust_name: RustIdent,
     pub proto_type: field_descriptor_proto::Type,
     wire_type: wire_format::WireType,
+    wire_name: RustIdent,
```
Comment (stepancheg) [inline protobuf-codegen/src/field/mod.rs]: It is not a Rust ident, it is serde ident. Can you please make it:

```
serde_name: String
```

?

File: protobuf-codegen/src/field/mod.rs [hunk 2]
```diff
@@ -1234,6 +1229,25 @@ impl<'a> FieldGen<'a> {
         }
     }
 
+    fn write_serde_attr(&self, w: &mut CodeWriter) {
+        let mut tags = vec![];
```
Comment (stepancheg) [inline protobuf-codegen/src/field/mod.rs]: `Vec::new()`

File: protobuf-codegen/src/field/mod.rs [hunk 3]
```diff
@@ -2605,4 +2619,4 @@ pub(crate) fn rust_field_name_for_protobuf_field_name(name: &str) -> RustIdent {
     } else {
         RustIdent::new(name)
     }
-}
+}
```
Comment (stepancheg) [inline protobuf-codegen/src/field/mod.rs]: Trailing newline is needed.

File: protobuf-test/src/common/v2/test_serde_derive_pb.proto [hunk 4]
```diff
@@ -48,4 +48,8 @@ message TestSerdeMap {
 message TestSerdeVec {
     option (rustproto.repeated_field_vec) = true;
     repeated string test_repeated = 1;
+}
+
+message TestRenameField {
+    optional string static = 1;
```
Comment (stepancheg) [inline protobuf-test/src/common/v2/test_serde_derive_pb.proto]: And here.

File: protobuf-test/src/common/v2/test_serde_derive.rs [hunk 5]
```diff
@@ -127,4 +127,10 @@ fn test_deserialize_with_missing_map() {
 fn test_deserialize_with_missing_repeated_vector() {
     let deserialized: TestSerdeVec = serde_json::from_str(&"{}").unwrap();
     assert_eq!(0, deserialized.test_repeated.len());
+}
+
+#[test]
+fn test_deserialize_with_renamed_field() {
+    let deserialized: TestRenameField = serde_json::from_str(&r#"{"static": "works"}"#).unwrap();
+    assert_eq!(deserialized.get_field_static(), "works");
 }
```
Comment (stepancheg) [inline protobuf-test/src/common/v2/test_serde_derive.rs]: And here

File: protobuf-codegen/src/field/mod.rs (hunk 1 above)
Comment (stepancheg) [reply protobuf-codegen/src/field/mod.rs]: ... or even `serde_alias`.

**Comment by calavera** [comment]
Thanks @samschlegel ! I think I've addressed all your comments. Do you prefer to preserve the git history or should I squash all those commits together? I don't have a preference.

**Comment by stepancheg** [comment]
Thanks!

I've squashed commits with github interface.

**Pull request merged**